model/flow/sealing_segment.go

```diff
@@ -274,46 +279,57 @@ func (builder *SealingSegmentBuilder) hasValidSeal() bool {
 		for _, seal := range block.Payload.Seals {
 			// if we found the latest seal, confirm it seals lowest
 			if seal.ID() == latestSealID {
-				return seal.BlockID == lowestID
+				if seal.BlockID == lowestID {
+					return seal, true
```

Comment:
validate not only validates the first block is sealed, but also return the seal, so that we can attach it to the created SealingSegment, for it to be used in building the index.